rest_framework/schemas.py

```diff
@@ -256,6 +256,289 @@ def get_allowed_methods(self, callback):
         ]
 
 
+class ViewInspector(object):
+    """
+    Descriptor class on APIView.
+
+    Provide subclass for per-view schema generation
+    """
+    def __get__(self, instance, owner):
+        self.view = instance
```

Comment:
I think this is a bit obscure. An alternative would be to expose `get_link` on the API View, that calls into the inspector, including the view.

Ie.

    # Schema generation   (Probably introduce this at L363)

    def get_link(self, path, method, base_url):
        """
        Return a `coreapi.Link()` for the given path template and method.

        This is the abstract representation that we use to build up a complete
        API Schema, that can be used for documentation and other tooling.
        """
        return self.schema.get_link(self, path, method, base_url)

Then use `view` as an explicit argument everywhere in the inspector methods, rather than an instance.

Reply:
This is just _How Python Descriptors Work_ — So, is it obscure? Well, yes, but it's just one of the lesser used parts of the programming model. It's just Python. (It's exactly how all our declarative DSLs are built.) 

The advantage (the whole point) of descriptors is that you get a reference back to the parent instance (or class) without needing to pass the extra reference by hand. You then get the nice familiar property dot-syntax to go with it. 

If we don't use a descriptor we have (here) **two** API additions to `APIView` instead of one. `schema` and `get_link` instead of just `schema`. 

I could add a docstring saying something about `ViewInspector is a descriptor [link to how to guide](https://docs.python.org/3/howto/descriptor.html)`. 

Who's looking at it? Two types of people:

1. End users: They're never meant to even see this. They just declare their `schema=ManualSchema(...)` or whatever and then use it like every other object property they've ever seen. (And in Django almost all of those are descriptors too.) 
2. DRF contributors: We can cope. 😀

If don't use a descriptor here I would pull it off of APIView entirely. Instead instantiate an instance with a view parameter in the clear: 

```
   inspector = AutoSchema(view)
   inspector.get_link(...)
```

But here we loose the ability to declare the customisations... so we can't just set a class on the view, we have to use an instance... so we might as well use the descriptor `__get__` method...

Reply:
> If don't use a descriptor here I would pull it off of APIView entirely. 

No, let's not go down that route.

Okay, alternate - let's just make sure that `__get__` has a docstring, noting the usage.

Reply:
👍

Reply:
aab4196b05e834326db179e919e3d955aaa0cbd1